ADV.LOGO

```diff
@@ -0,0 +1,346 @@
+50 PR "OK"
+60 MAKE "RNUM" :TRYMOVE
+70 LOOK
+END
+
+; Special actions
+
+TO WAVE :ITEM
+10 IF NOT IHAVE? :ITEM PR SE [YOU ARE HOLDING NO] PERIOD :ITEM CMD
+20 IF NOT :ITEM = "WAND" NOTHING
+30 IF AND NOT :RNUM = 4 NOT :RNUM = 5 PR [NOTHING HAPPENS.] CMD
+40 PR [POOF THE SCENE CHANGES]
+50 IF :RNUM = 4 MAKE "RNUM" 5 ELSE MAKE "RNUM" 4
+60 LOOK
+END
+
+TO A.FIX :ITEM					;FIX originally. conflicts with built-in procedrue
+10 IF IHAVE? :ITEM PR [YOU HAVE TO DROP IT TO FIX IT] CMD
+20 IF NOT HERE? :ITEM SEENO :ITEM
+30 IF NOT :ITEM = "MACHINE" PR [YOU CANT FIX THAT] CMD
+40 IF NOT ITEMLOC "WAND" = 0 PR [THE MACHINE IS NOT BROKEN] CMD
+50 IF NOT IHAVE? "SCREWDRIVER" PR [YOU DONT HAVE THE PRPPER TOOLS TO] PR [FIX IT] CMD
+60 PR [YOU FIX THE MACHINE WITH YOUR TRUSTY]
+70 PR [SCREWDRIVER. UPON BEING FIXED THE]
+80 PR [MACHINE STARTS UP AND PRODUCES A WAND]
+90 PUTITEM "WAND" 4
+100 CMD
+END
+
+TO A.EXAMINE :ITEM
+10 IF NOT HERE? :ITEM  PR [I SEE NO] :ITEM [HERE]  CMD
+20 IF :ITEM = "WAND" PR [IT BEARS A FADED INSCRIPTION:] PR [WAVE ME AND YOU'LL BE GLAD.] CMD
+30 IF NOT :ITEM = "MACHINE" PR SE [I SEE NOTHING SPECIAL ABOUT THE] PERIOD :ITEM CMD
+40 IF NOT 0 = ITEMLOC "WAND" PR [IT SEEMS TO BEAR THE MARKS OF A HASTY] PR [REPAIR JOB.] CMD
+50 PR [IT IS BROKEN YOU COULD FIX IT WITH]
+60 PR [THE RIGHT TOOL]
+70 CMD
+END
+
+;; Items inventory, cmd, help, score procedures section
+
+TO PITEMS :LOC                           
+10 PITEMS2 :LOC :ITEMS	
+END
+
+TO PITEMS2 :LOC :I
+10 IF :I = [] STOP
+20 IF FIRST FIRST :I = :LOC PRINT LAST FIRST :I
+30 PITEMS2 :LOC BF :I
+END
+
+TO CMD
+10 CR
+20 TYPE "COMMAND"
+ ; toplevel isn't a command in LLOGO but rather an operation 
+; and returns value. So we cannot use it as originally intended,
+; so we should use STOP after CMD invocation to interrupt
+; evaluation process.
+;30 TOPLEVEL
+END
+
+TO GAME
+10 OP "ADVSAVE"
+END
+
+TO ITEMLOC2 :ITEM :I
+10 IF :I = [] OP 0
+20 IF LAST FIRST :I = :ITEM OP FIRST FIRST :I
+30 OP ITEMLOC2 :ITEM BF :I
+END
+
+TO PUTITEM :ITEM :LOC
+10 MAKE "ITEMS" PUTITEM2 :ITEM :LOC :ITEMS
+END
+
+TO PUTITEM2 :ITEM :LOC :LIST
+10 IF :LIST = [] OP []
+20 IF LAST FIRST :LIST = :ITEM OP FPUT FPUT :LOC BF FIRST :LIST BF :LIST
+30 OP FPUT FIRST :LIST PUTITEM2 :ITEM :LOC BF :LIST
+END
+
+
+TO SETIT :THING
+10 MAKE "IT" :THING
+20 OP :THING
+END
+
+TO SEENO :I
+10 PR SE [I SEE NO] SE :I "HERE"
+20 CMD
+END
+
+TO EVERYTHING
+10 OP "EVERYTHING"
+END
+
+TO IHAVE? :ITEM
+10 OP EQUAL (- 1) ITEMLOC :ITEM
+END
+
+TO ITEMLOC :ITEM
+10 OP ITEMLOC2 :ITEM :ITEMS
+END
+
+TO PERIOD :WORD
+10 OP WORD :WORD "."
+END
+
+TO IT
+10 OP :IT
+END
+
+TO HERE? :ITEM
+10 LOCAL "LOC"
+20 MAKE "LOC" ITEMLOC :ITEM
+30 OP OR (- 1) = :LOC :RNUM = :LOC
+END
+
+TO INVENT
+10 PITEMS (- 1)
+20 CMD
+END
+
+TO HELP
+100 PR [TO MOVE TYPE]
+110 PR [NORTH EAST SOUTH WEST]
+120 PR [FOR NORTH SOUTH EAST WEST]
+130 CR
+140 PR [TYPE LOOK TO SEE WHAT ROOM YOU]
+150 PR [ARE IN. YOU CAN TAKE AND DROP ITEMS]
+160 PR [INVENT SHOWS YOUR INVENTORY]
```

Comment:
End sentence with period.

Reply:
Actually it would be better to phrase it as a call to action:
TYPE INVENT TO SHOW YOUR INVENTORY.